### DISCUSSION TITLE
Populate Options-picker based on values selected in Multi-select Picker

### DISCUSSION THREAD

**Discussion by Simonkcaon** [Budibase/budibase#7449] [Help] [answered]
**What i want**: I have a list of Items and in the list the user is supposed to pick the most relevant for him. (multi-select)

After he selects the most relevant Items, only the items selected in the multi-select are shown in an options-picker where he is supposed to select the **one** most relevant to him.

**Problem:**
I'm struggling to populate the options-picker with all the selected items from the multi-select. Both pickers get the data from one table. 
I tried putting the options-picker inside a data provider, filter by form field of the multi-select. This works, but only for one item. 

I'm a bit stuck here and would appreciate some help. 
Thank you :)

**Comment by R2bEEaton** [comment, marked as answer]
Hi @Simonkcaon,

You were very close with this idea!
> I tried putting the options-picker inside a data provider, filter by form field of the multi-select. This works, but only for one item.

### How to do it!

To achieve your solution, create the following structure:

![image](https://user-images.githubusercontent.com/34921506/186649435-ea7ce902-2928-4e84-b0b0-221889466fcb.png)

1. `New Data Provider` should be the source of your options.
2. `New Multi-select Picker` should take its options from `New Data Provider`. Set the Label Column to your desired label, and the value column to `_id`. Set the field to something like `multi`.
3. `New Data Provider 2` should be the same source of your options, with the following filter: `_id` `Is in` `Binding` `{{ New Form.Fields.multi }}`. The **Is in** is the crucial part to make your idea work.
4. `New Options Picker` should take its options from `New Data Provider 2` and follow the same label and value scheme for easier use later.
5. Now you can add a button or whatever you want to submit the form!

Here's an export if you want to examine this further: [MultiPicker-export-1661426335138.txt](https://github.com/Budibase/budibase/files/9424090/MultiPicker-export-1661426335138.txt)


### Enjoy!
Hope this helped. Reply with any further questions or accept this answer using the checkmark above to help future readers!

**Reply by Simonkcaon** [reply]
**Thank you so much!** It worked. The only problem I'm facing now is that even tho it works it throws me an Error: 

"You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') order by `multi`.`id` desc limit 50) as `multi` order by `u...' at line 1"

This only happens when filtering with " is in". Not sure if it's a problem with our MariaDB setup or the way "is in" is handled in budibase.

**Reply by Simonkcaon** [reply]
Fixed the problem. Had to set a condition that sets the filter only if "Form.Field.multi" is not empty.

**Reply by R2bEEaton** [reply]
Awesome!
